<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Input formats and creation of a REVOLVER cohort • revolver</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Input formats and creation of a REVOLVER cohort">
<meta property="og:description" content="">
<meta property="og:image" content="caravagn.github.io/revolver/logo.png">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">revolver</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.3.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/revolver.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/FAQ.html">Frequently Asked Questions</a>
    </li>
    <li>
      <a href="../articles/Getters.html">Getter functions</a>
    </li>
    <li>
      <a href="../articles/Input_formats.html">Input formats and creation of a REVOLVER cohort</a>
    </li>
    <li>
      <a href="../articles/Plotting_functions.html">Plotting functions</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/caravagn/revolver">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Input formats and creation of a REVOLVER cohort</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/caravagn/revolver/blob/master/vignettes/Input_formats.Rmd"><code>vignettes/Input_formats.Rmd</code></a></small>
      <div class="hidden name"><code>Input_formats.Rmd</code></div>

    </div>

    
    
<div id="input-format-for-revolver" class="section level2">
<h2 class="hasAnchor">
<a href="#input-format-for-revolver" class="anchor"></a>Input format for REVOLVER</h2>
<p>Data from a cohort of patients can be represented as a <em>dataframe</em> with at least 7 columns, where every row represents one genomic alteration annotated for the analysis.</p>
<table class="table">
<colgroup>
<col width="33%">
<col width="33%">
<col width="33%">
</colgroup>
<thead><tr class="header">
<th>Field</th>
<th>Type</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><code>Misc</code></td>
<td>string</td>
<td><em>Customary annotation which is never used to analyse the data, but that might be good to carry around</em></td>
</tr>
<tr class="even">
<td><code>patientID</code></td>
<td>string</td>
<td><em>A patient identifier</em></td>
</tr>
<tr class="odd">
<td><code>variantID</code></td>
<td>string</td>
<td><em>An alteration identifier</em></td>
</tr>
<tr class="even">
<td><code>cluster</code></td>
<td>string</td>
<td><em>Group ID (eg., a clone, with CCF data)</em></td>
</tr>
<tr class="odd">
<td><code>is.driver</code></td>
<td>logical</td>
<td>TRUE <em>if the alteration is a driver to correlate when repeated trajectores are extracted</em>
</td>
</tr>
<tr class="even">
<td><code>is.clonal</code></td>
<td>logical</td>
<td>TRUE <em>if the alteration belongs to the group of clonal (truncal) events; there should only one such clonal group per patient</em>
</td>
</tr>
<tr class="odd">
<td><code>CCF</code></td>
<td>string</td>
<td><em>A parsable format for storage of input CCFs or binary data (see below)</em></td>
</tr>
</tbody>
</table>
<p>All identifiers cannot contain spaces or dash/ hyphen (<code><a href="https://rdrr.io/r/base/Arithmetic.html">-</a></code>) symbols, and must be unique:</p>
<ul>
<li><p>the <code>patientID</code> must be unique across patients;</p></li>
<li><p>the <code>variantID</code> for driver events (those with <code>is.driver = TRUE</code>) must be shared across all patients that have the same driver event, and cannot appear multiple times in the data of a single patient. Notice that you should not annotate an alteration as driver if it is not found in multiple patients.</p></li>
<li><p>the <code>variantID</code> for non-driver events (those with <code>is.driver = FALSE</code>) is in practice never used and therefore can be set freely.</p></li>
</ul>
<p>Notice that the input dataframe has the same structure for both CCF and binary input data.</p>
<div id="supported-alterations" class="section level3">
<h3 class="hasAnchor">
<a href="#supported-alterations" class="anchor"></a>Supported alterations</h3>
<p>The scope of supported alterations is broad, and include any SNV, larger chromosomal re-arrangment or other covariate that can be encoded in CCF or binary format.</p>
<p>The <code>variantID</code> field of driver alterations (<code>is.driver=TRUE</code>) will be matched to correlate trajectories in multiple patients, and can be you whatever you find more suitable for your analysis, for instance:</p>
<ul>
<li><p>a Hugo_Symbol (<code>BRAF</code>)</p></li>
<li><p>a name for a well-known SNV (<code>BRAF_600E</code>)</p></li>
<li><p>a reference to some cytoband (<code>3q26.32</code>)</p></li>
<li><p>your custom annotation (<code>MyFavoritePathway</code>).</p></li>
</ul>
<p>Alterations are also associated to groups (via <code>cluster</code>), which constitute the nodes of the computed trees. A group can have 0 drivers annotated, but every patient should have at least one driver to be analyzed with <code>REVOLVER</code>.</p>
<p>See also the <a href="https://caravagn.github.io/revolver/articles/FAQ.html">Frequently Asked Questions</a> if you are interested in modelling parallel evolution.</p>
</div>
<div id="input-values-for-ccf-or-binary-datasets" class="section level3">
<h3 class="hasAnchor">
<a href="#input-values-for-ccf-or-binary-datasets" class="anchor"></a>Input values for CCF or binary datasets</h3>
<p>Field <code>CCF</code> represents data from both types of <a href="https://caravagn.github.io/revolver/articles/FAQ.html">supported datasets</a>:</p>
<ul>
<li><p>real-valued CCF values (in [0, 1]),</p></li>
<li><p>or input binary values (either 0 or 1).</p></li>
</ul>
<p>Since patients can have different number of samples/ regions associated, <code>CCF</code> is a general string.</p>
<p>The format that we propose is simple, and easy to parse it:</p>
<pre><code># Example string that reports a CCF value 0.86 in region R1, 1 in R2 etc. 
R1:0.86;R2:1;R3:1

# Similarly, in binary presence/ absence format ... 
R1:0;R2:1;R3:1</code></pre>
<p>If you use this format, <code>REVOLVER</code> provides you a <a href="https://caravagn.github.io/revolver/reference/CCF_parser.html">parsing function</a> that you can pass to the <a href="https://caravagn.github.io/revolver/reference/revolver_cohort.html">cohort construction function</a></p>
</div>
<div id="example-data" class="section level3">
<h3 class="hasAnchor">
<a href="#example-data" class="anchor"></a>Example data</h3>
<p>An example cohort is released in the <code>evoverse.datasets</code> <a href="https://github.com/caravagn/evoverse.datasets">R package</a>, which also provides a <code>REVOLVER</code> objectwith the final analysis of the input cohort.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="co"># Load the package and runs available_data() to print what is available</span></a>
<a class="sourceLine" id="cb2-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(evoverse.datasets)</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-4" data-line-number="4"><span class="co">#&gt; Available datasets ~ use data('xxx', package='REVOLVER_datasets') to load dataset 'xxx'</span></a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb2-6" data-line-number="6"><span class="co">#&gt; ◉ TRACERx_NEJM_2017 Mutations from TRACERx lung (NEJM2017, PMID: 28445112). n = 99 patients, multi-region WES. </span></a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="co">#&gt; ◉ TRACERx_NEJM_2017_REVOLVER REVOLVER analysis of TRACERx_NEJM_2017.</span></a>
<a class="sourceLine" id="cb2-8" data-line-number="8"></a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="co"># Load an example dataset - the TRACERx cohort</span></a>
<a class="sourceLine" id="cb2-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TRACERx_NEJM_2017"</span>, <span class="dt">package =</span> <span class="st">"evoverse.datasets"</span>)</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"></a>
<a class="sourceLine" id="cb2-12" data-line-number="12"><span class="co"># This is a dataset that follows the specifications given above</span></a>
<a class="sourceLine" id="cb2-13" data-line-number="13"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(TRACERx_NEJM_<span class="dv">2017</span>)</a>
<a class="sourceLine" id="cb2-14" data-line-number="14"><span class="co">#&gt; # A tibble: 65,421 x 7</span></a>
<a class="sourceLine" id="cb2-15" data-line-number="15"><span class="co">#&gt;    Misc          patientID variantID cluster is.driver is.clonal CCF       </span></a>
<a class="sourceLine" id="cb2-16" data-line-number="16"><span class="co">#&gt;    &lt;chr&gt;         &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;   &lt;lgl&gt;     &lt;lgl&gt;     &lt;chr&gt;     </span></a>
<a class="sourceLine" id="cb2-17" data-line-number="17"><span class="co">#&gt;  1 CRUK0001:7:4… CRUK0001  PKD1L1    1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-18" data-line-number="18"><span class="co">#&gt;  2 CRUK0001:21:… CRUK0001  PCNT      1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-19" data-line-number="19"><span class="co">#&gt;  3 CRUK0001:22:… CRUK0001  CSF2RB    1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-20" data-line-number="20"><span class="co">#&gt;  4 CRUK0001:10:… CRUK0001  KAZALD1   1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-21" data-line-number="21"><span class="co">#&gt;  5 CRUK0001:7:8… CRUK0001  CACNA2D1  1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-22" data-line-number="22"><span class="co">#&gt;  6 CRUK0001:1:2… CRUK0001  IPO9      1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-23" data-line-number="23"><span class="co">#&gt;  7 CRUK0001:3:1… CRUK0001  PDCD10    1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-24" data-line-number="24"><span class="co">#&gt;  8 CRUK0001:7:1… CRUK0001  CALD1     1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-25" data-line-number="25"><span class="co">#&gt;  9 CRUK0001:17:… CRUK0001  CANT1     1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-26" data-line-number="26"><span class="co">#&gt; 10 CRUK0001:16:… CRUK0001  IFT140    1       FALSE     FALSE     R1:0.86;R…</span></a>
<a class="sourceLine" id="cb2-27" data-line-number="27"><span class="co">#&gt; # … with 65,411 more rows</span></a></code></pre></div>
</div>
</div>
<div id="cohort-construction" class="section level2">
<h2 class="hasAnchor">
<a href="#cohort-construction" class="anchor"></a>Cohort construction</h2>
<p>Once you have a dataframe/ tibble in the required format, you can use a <a href="https://caravagn.github.io/revolver/reference/revolver_cohort.html">cohort construction function</a>.</p>
<p>In this case, to make it faster we only use 3 patients from the released cohort.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(tidyverse)</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/base/library.html">require</a></span>(revolver)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="co">#&gt; Loading required package: revolver</span></a>
<a class="sourceLine" id="cb3-4" data-line-number="4"><span class="co">#&gt; Warning: replacing previous import 'igraph::%s%' by 'matrixcalc::%s%' when</span></a>
<a class="sourceLine" id="cb3-5" data-line-number="5"><span class="co">#&gt; loading 'ctree'</span></a>
<a class="sourceLine" id="cb3-6" data-line-number="6"><span class="co">#&gt; Warning: replacing previous import 'igraph::groups' by 'tidygraph::groups'</span></a>
<a class="sourceLine" id="cb3-7" data-line-number="7"><span class="co">#&gt; when loading 'ctree'</span></a>
<a class="sourceLine" id="cb3-8" data-line-number="8"><span class="co">#&gt;  [ ctree - Clone Trees in cancer ] </span></a>
<a class="sourceLine" id="cb3-9" data-line-number="9"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb3-10" data-line-number="10"><span class="co">#&gt; GitHub :  caravagn/ctree</span></a>
<a class="sourceLine" id="cb3-11" data-line-number="11"><span class="co">#&gt; Warning: replacing previous import 'dendextend::theme_dendro' by</span></a>
<a class="sourceLine" id="cb3-12" data-line-number="12"><span class="co">#&gt; 'ggdendro::theme_dendro' when loading 'revolver'</span></a>
<a class="sourceLine" id="cb3-13" data-line-number="13"><span class="co">#&gt; Warning: replacing previous import 'dendextend::rotate' by 'ggpubr::rotate'</span></a>
<a class="sourceLine" id="cb3-14" data-line-number="14"><span class="co">#&gt; when loading 'revolver'</span></a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="co">#&gt; Warning: replacing previous import 'igraph::groups' by 'tidygraph::groups'</span></a>
<a class="sourceLine" id="cb3-16" data-line-number="16"><span class="co">#&gt; when loading 'revolver'</span></a>
<a class="sourceLine" id="cb3-17" data-line-number="17"><span class="co">#&gt;  [ REVOLVER - Repeated Evolution in Cancer ] </span></a>
<a class="sourceLine" id="cb3-18" data-line-number="18"><span class="co">#&gt; Author :  Giulio Caravagna &lt;gcaravagn@gmail.com&gt; </span></a>
<a class="sourceLine" id="cb3-19" data-line-number="19"><span class="co">#&gt; GitHub :  caravagn/revolver</span></a>
<a class="sourceLine" id="cb3-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-21" data-line-number="21"><span class="co">#&gt; Attaching package: 'revolver'</span></a>
<a class="sourceLine" id="cb3-22" data-line-number="22"><span class="co">#&gt; The following object is masked from 'package:igraph':</span></a>
<a class="sourceLine" id="cb3-23" data-line-number="23"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-24" data-line-number="24"><span class="co">#&gt;     plot_dendrogram</span></a>
<a class="sourceLine" id="cb3-25" data-line-number="25"></a>
<a class="sourceLine" id="cb3-26" data-line-number="26">subset_data =<span class="st"> </span>TRACERx_NEJM_<span class="dv">2017</span> <span class="op">%&gt;%</span></a>
<a class="sourceLine" id="cb3-27" data-line-number="27"><span class="st">  </span><span class="kw"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span>(patientID <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"CRUK0001"</span>, <span class="st">"CRUK0002"</span>, <span class="st">"CRUK0003"</span>))</a>
<a class="sourceLine" id="cb3-28" data-line-number="28"></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="co"># Create a cohort</span></a>
<a class="sourceLine" id="cb3-30" data-line-number="30">my_cohort =<span class="st"> </span><span class="kw"><a href="../reference/revolver_cohort.html">revolver_cohort</a></span>(</a>
<a class="sourceLine" id="cb3-31" data-line-number="31">  subset_data, </a>
<a class="sourceLine" id="cb3-32" data-line-number="32">  <span class="dt">CCF_parser =</span> CCF_parser,</a>
<a class="sourceLine" id="cb3-33" data-line-number="33">  <span class="dt">ONLY.DRIVER =</span> <span class="ot">FALSE</span>, </a>
<a class="sourceLine" id="cb3-34" data-line-number="34">  <span class="dt">MIN.CLUSTER.SIZE =</span> <span class="dv">0</span>,</a>
<a class="sourceLine" id="cb3-35" data-line-number="35">  <span class="dt">annotation =</span> <span class="st">"My small REVOLVER dataset"</span></a>
<a class="sourceLine" id="cb3-36" data-line-number="36">  )</a>
<a class="sourceLine" id="cb3-37" data-line-number="37"><span class="co">#&gt;  [ REVOLVER Cohort constructor ] </span></a>
<a class="sourceLine" id="cb3-38" data-line-number="38"><span class="co">#&gt; ◉  Use only drivers 1 </span></a>
<a class="sourceLine" id="cb3-39" data-line-number="39"><span class="co">#&gt; ◉ Reject clusters with size below 0 </span></a>
<a class="sourceLine" id="cb3-40" data-line-number="40"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-41" data-line-number="41"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-42" data-line-number="42"><span class="co">#&gt;  REVOLVER input data</span></a>
<a class="sourceLine" id="cb3-43" data-line-number="43"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-44" data-line-number="44"><span class="co">#&gt; # A tibble: 2,710 x 9</span></a>
<a class="sourceLine" id="cb3-45" data-line-number="45"><span class="co">#&gt;    Misc  patientID variantID cluster is.driver is.clonal CCF   id   </span></a>
<a class="sourceLine" id="cb3-46" data-line-number="46"><span class="co">#&gt;    &lt;chr&gt; &lt;chr&gt;     &lt;chr&gt;     &lt;chr&gt;   &lt;lgl&gt;     &lt;lgl&gt;     &lt;chr&gt; &lt;chr&gt;</span></a>
<a class="sourceLine" id="cb3-47" data-line-number="47"><span class="co">#&gt;  1 CRUK… CRUK0001  PKD1L1    1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-48" data-line-number="48"><span class="co">#&gt;  2 CRUK… CRUK0001  PCNT      1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-49" data-line-number="49"><span class="co">#&gt;  3 CRUK… CRUK0001  CSF2RB    1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-50" data-line-number="50"><span class="co">#&gt;  4 CRUK… CRUK0001  KAZALD1   1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-51" data-line-number="51"><span class="co">#&gt;  5 CRUK… CRUK0001  CACNA2D1  1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-52" data-line-number="52"><span class="co">#&gt;  6 CRUK… CRUK0001  IPO9      1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-53" data-line-number="53"><span class="co">#&gt;  7 CRUK… CRUK0001  PDCD10    1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-54" data-line-number="54"><span class="co">#&gt;  8 CRUK… CRUK0001  CALD1     1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-55" data-line-number="55"><span class="co">#&gt;  9 CRUK… CRUK0001  CANT1     1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-56" data-line-number="56"><span class="co">#&gt; 10 CRUK… CRUK0001  IFT140    1       FALSE     FALSE     R1:0… __mu…</span></a>
<a class="sourceLine" id="cb3-57" data-line-number="57"><span class="co">#&gt; # … with 2,700 more rows, and 1 more variable: cluster_size &lt;int&gt;</span></a>
<a class="sourceLine" id="cb3-58" data-line-number="58"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-59" data-line-number="59"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-60" data-line-number="60"><span class="co">#&gt;  Extracting dataset for each patient (this may take some time)</span></a>
<a class="sourceLine" id="cb3-61" data-line-number="61"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-62" data-line-number="62"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-63" data-line-number="63"><span class="co">#&gt; CRUK0001 2100 entries </span></a>
<a class="sourceLine" id="cb3-64" data-line-number="64"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-65" data-line-number="65"><span class="co">#&gt; CRUK0002 280 entries </span></a>
<a class="sourceLine" id="cb3-66" data-line-number="66"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-67" data-line-number="67"><span class="co">#&gt; CRUK0003 330 entries </span></a>
<a class="sourceLine" id="cb3-68" data-line-number="68"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-69" data-line-number="69"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-70" data-line-number="70"><span class="co">#&gt;  Extracting clones' table for each patient</span></a>
<a class="sourceLine" id="cb3-71" data-line-number="71"><span class="co">#&gt; =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-</span></a>
<a class="sourceLine" id="cb3-72" data-line-number="72"><span class="co">#&gt; CRUK0001 2100 entries  --&gt;  11 clones </span></a>
<a class="sourceLine" id="cb3-73" data-line-number="73"><span class="co">#&gt; CRUK0002 280 entries  --&gt;  7 clones </span></a>
<a class="sourceLine" id="cb3-74" data-line-number="74"><span class="co">#&gt; CRUK0003 330 entries  --&gt;  7 clones</span></a>
<a class="sourceLine" id="cb3-75" data-line-number="75"></a>
<a class="sourceLine" id="cb3-76" data-line-number="76"><span class="co"># This is now a REVOLVER cohort with S3 methods available</span></a>
<a class="sourceLine" id="cb3-77" data-line-number="77"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(my_cohort)</a>
<a class="sourceLine" id="cb3-78" data-line-number="78"><span class="co">#&gt;  [ REVOLVER - Repeated Evolution in Cancer ] </span></a>
<a class="sourceLine" id="cb3-79" data-line-number="79"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-80" data-line-number="80"><span class="co">#&gt; Dataset : My small REVOLVER dataset </span></a>
<a class="sourceLine" id="cb3-81" data-line-number="81"><span class="co">#&gt; Cohort  : 3 patients, 2710 variants and 16 driver events. </span></a>
<a class="sourceLine" id="cb3-82" data-line-number="82"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-83" data-line-number="83"><span class="co">#&gt; Trees per patient    : NO </span></a>
<a class="sourceLine" id="cb3-84" data-line-number="84"><span class="co">#&gt; Fit via TL           : NO </span></a>
<a class="sourceLine" id="cb3-85" data-line-number="85"><span class="co">#&gt; REVOLVER clustering  : NO </span></a>
<a class="sourceLine" id="cb3-86" data-line-number="86"><span class="co">#&gt; Jackknife statistics : NO </span></a>
<a class="sourceLine" id="cb3-87" data-line-number="87"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb3-88" data-line-number="88"><span class="co">#&gt; For summary statistics see `?Stats_*(x)` with * = {cohort, drivers, trees, fits, clusters, ...} </span></a>
<a class="sourceLine" id="cb3-89" data-line-number="89"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-90" data-line-number="90"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-91" data-line-number="91"><span class="co">#&gt;  ERROR  Some driver variantIDs occur only once and should therefore be removed. </span></a>
<a class="sourceLine" id="cb3-92" data-line-number="92"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb3-93" data-line-number="93"><span class="co">#&gt;  You can use `revolver::Stats_drivers(x)` to retrieve them. </span></a>
<a class="sourceLine" id="cb3-94" data-line-number="94"><span class="co">#&gt; # A tibble: 14 x 7</span></a>
<a class="sourceLine" id="cb3-95" data-line-number="95"><span class="co">#&gt;    variantID numClonal p_clonal numSubclonal p_subclonal N_tot p_tot</span></a>
<a class="sourceLine" id="cb3-96" data-line-number="96"><span class="co">#&gt;    &lt;chr&gt;         &lt;dbl&gt;    &lt;dbl&gt;        &lt;dbl&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></a>
<a class="sourceLine" id="cb3-97" data-line-number="97"><span class="co">#&gt;  1 TP53              1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-98" data-line-number="98"><span class="co">#&gt;  2 MGA               1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-99" data-line-number="99"><span class="co">#&gt;  3 WRN               1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-100" data-line-number="100"><span class="co">#&gt;  4 MET               1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-101" data-line-number="101"><span class="co">#&gt;  5 TERT              1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-102" data-line-number="102"><span class="co">#&gt;  6 PIK3CA            1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-103" data-line-number="103"><span class="co">#&gt;  7 CDKN2A            1    0.333            0       0         1 0.333</span></a>
<a class="sourceLine" id="cb3-104" data-line-number="104"><span class="co">#&gt;  8 ARHGAP35          0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-105" data-line-number="105"><span class="co">#&gt;  9 PASK              0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-106" data-line-number="106"><span class="co">#&gt; 10 RB1               0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-107" data-line-number="107"><span class="co">#&gt; 11 IKZF1             0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-108" data-line-number="108"><span class="co">#&gt; 12 KRAS              0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-109" data-line-number="109"><span class="co">#&gt; 13 EP300             0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-110" data-line-number="110"><span class="co">#&gt; 14 CTNNB1            0    0                1       0.333     1 0.333</span></a>
<a class="sourceLine" id="cb3-111" data-line-number="111"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(my_cohort)</a></code></pre></div>
<p><img src="Input_formats_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<p>The cohort is now ready to be inspected and analysed (note that since we only used 3 patients a warning is raised to mention that the cohort contains drivers annotated which are not really recurrent and that, therefore, should be removed; if you construct the cohort object with the full set of 99 patients that warning will vanish).</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1"><span class="co"># Load a full cohort - the TRACERx cohort</span></a>
<a class="sourceLine" id="cb4-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"TRACERx_NEJM_2017_REVOLVER"</span>, <span class="dt">package =</span> <span class="st">"evoverse.datasets"</span>)</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"></a>
<a class="sourceLine" id="cb4-4" data-line-number="4"><span class="co"># We can use S3 object functions to retrieve simple information about the cohort.</span></a>
<a class="sourceLine" id="cb4-5" data-line-number="5"><span class="co"># The `print` functions runs also the `revolver_check_cohort` function which </span></a>
<a class="sourceLine" id="cb4-6" data-line-number="6"><span class="co"># tells us that some patient have only 1 clone with drivers, and therefore they</span></a>
<a class="sourceLine" id="cb4-7" data-line-number="7"><span class="co"># can just be expanded.</span></a>
<a class="sourceLine" id="cb4-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/base/print.html">print</a></span>(TRACERx_NEJM_<span class="dv">2017</span>_REVOLVER)</a>
<a class="sourceLine" id="cb4-9" data-line-number="9"><span class="co">#&gt;  [ REVOLVER - Repeated Evolution in Cancer ] </span></a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-11" data-line-number="11"><span class="co">#&gt; Dataset : TRACERx NEJM 2017 </span></a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co">#&gt; Cohort  : 99 patients, 450 variants and 79 driver events. </span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-14" data-line-number="14"><span class="co">#&gt; Trees per patient    : YES </span></a>
<a class="sourceLine" id="cb4-15" data-line-number="15"><span class="co">#&gt; Fit via TL           : YES </span></a>
<a class="sourceLine" id="cb4-16" data-line-number="16"><span class="co">#&gt; REVOLVER clustering  : YES </span></a>
<a class="sourceLine" id="cb4-17" data-line-number="17"><span class="co">#&gt; Jackknife statistics : YES </span></a>
<a class="sourceLine" id="cb4-18" data-line-number="18"><span class="co">#&gt;  </span></a>
<a class="sourceLine" id="cb4-19" data-line-number="19"><span class="co">#&gt; For summary statistics see `?Stats_*(x)` with * = {cohort, drivers, trees, fits, clusters, ...} </span></a>
<a class="sourceLine" id="cb4-20" data-line-number="20"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-21" data-line-number="21"><span class="co">#&gt; </span></a>
<a class="sourceLine" id="cb4-22" data-line-number="22"><span class="co">#&gt;  </span><span class="al">WARNING</span><span class="co">  Some patients have only one clone with drivers, and therefore they will just be expanded. </span></a>
<a class="sourceLine" id="cb4-23" data-line-number="23"><span class="co">#&gt; # A tibble: 54 x 7</span></a>
<a class="sourceLine" id="cb4-24" data-line-number="24"><span class="co">#&gt;    patientID numBiopsies numMutations numDriverMutati… numClonesWithDr…</span></a>
<a class="sourceLine" id="cb4-25" data-line-number="25"><span class="co">#&gt;    &lt;chr&gt;           &lt;int&gt;        &lt;int&gt;            &lt;int&gt;            &lt;int&gt;</span></a>
<a class="sourceLine" id="cb4-26" data-line-number="26"><span class="co">#&gt;  1 CRUK0007            2            3                3                1</span></a>
<a class="sourceLine" id="cb4-27" data-line-number="27"><span class="co">#&gt;  2 CRUK0010            2            3                3                1</span></a>
<a class="sourceLine" id="cb4-28" data-line-number="28"><span class="co">#&gt;  3 CRUK0012            2            1                1                1</span></a>
<a class="sourceLine" id="cb4-29" data-line-number="29"><span class="co">#&gt;  4 CRUK0018            4            4                4                1</span></a>
<a class="sourceLine" id="cb4-30" data-line-number="30"><span class="co">#&gt;  5 CRUK0019            2            1                1                1</span></a>
<a class="sourceLine" id="cb4-31" data-line-number="31"><span class="co">#&gt;  6 CRUK0021            2            4                4                1</span></a>
<a class="sourceLine" id="cb4-32" data-line-number="32"><span class="co">#&gt;  7 CRUK0025            3            3                3                1</span></a>
<a class="sourceLine" id="cb4-33" data-line-number="33"><span class="co">#&gt;  8 CRUK0026            2            4                4                1</span></a>
<a class="sourceLine" id="cb4-34" data-line-number="34"><span class="co">#&gt;  9 CRUK0028            2            2                2                1</span></a>
<a class="sourceLine" id="cb4-35" data-line-number="35"><span class="co">#&gt; 10 CRUK0029            6            4                4                1</span></a>
<a class="sourceLine" id="cb4-36" data-line-number="36"><span class="co">#&gt; # … with 44 more rows, and 2 more variables: numTruncalMutations &lt;int&gt;,</span></a>
<a class="sourceLine" id="cb4-37" data-line-number="37"><span class="co">#&gt; #   numSubclonalMutations &lt;int&gt;</span></a>
<a class="sourceLine" id="cb4-38" data-line-number="38"></a>
<a class="sourceLine" id="cb4-39" data-line-number="39"><span class="co"># Plot</span></a>
<a class="sourceLine" id="cb4-40" data-line-number="40"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(TRACERx_NEJM_<span class="dv">2017</span>_REVOLVER)</a></code></pre></div>
<p><img src="Input_formats_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#input-format-for-revolver">Input format for REVOLVER</a></li>
      <li><a href="#cohort-construction">Cohort construction</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by <a href="https://sites.google.com/site/giuliocaravagna/">Giulio Caravagna</a>.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.0.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
